src/connection/host_lookup.c

```diff
@@ -686,6 +686,22 @@ getaddrinfo_uhost (char *node, char *service, struct addrinfo *hints, struct add
   return ret;
 }
 
+void
+freeaddrinfo_uhost (struct addrinfo *res)
+{
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+      return (freeaddrinfo (res));
+    }
+
+  if (res)
+    {
+      FREE_MEM (res->ai_canonname);
```

Comment:
It is good to me to replace malloc to MALLOC() macro if you want to use FREE_MEM() macro. Or how about use free()?

Reply:
I think it's a good idea to use MALLOC and FREE_MEM as a pair in point of memory managements.
fixed.